requests_cache/backends/base.py

```diff
@@ -92,8 +92,12 @@ def save_response(self, response: AnyResponse, cache_key: str = None, expires: d
             expire_after: Time in seconds until this cache item should expire
         """
         cache_key = cache_key or self.create_key(response.request)
-        cached_response = CachedResponse.from_response(response, expires=expires)
-        cached_response = redact_response(cached_response, self.ignored_parameters)
+        if isinstance(response, CachedResponse):
```

Comment:
Could we keep this section the way it is, and put this check in `CachedResponse.from_response` instead? E.g.:
```python
    @classmethod
    def from_response(cls, original_response: Response, expires: datetime = None, **kwargs):
        """Create a CachedResponse based on an original response object"""
        if isinstance(original_response, CachedResponse):
            original_response.expires = expires
            return original_response

        obj = cls(expires=expires, **kwargs)

        ...
```

Reply:
Done. Although I opted to return a copy of the original_response instead of the same object. I guess the user would not expect to be served the same instance.